pages/mesosphere/dcos/1.14/installing/production/advanced-configuration/configuration-reference/index.md

```diff
@@ -248,6 +250,18 @@ Use this to set up a custom CA certificate. See [using a Custom CA Certificate](
 
 Use this to set up a custom CA certificate. See [using a Custom CA Certificate](/mesosphere/dcos/1.14/security/ent/tls-ssl/ca-custom#configuration-parameter-reference) page for a detailed configuration parameter reference. 
 
+### exhibitor_tls_enabled [enterprise type="inline" size="small" /]
+
+When `true` (default), the DC/OS installer will automatically lock-down the exhibitor service using X509 end entity certificates signed by a DC/OS global, unique CA certificate. See [Exhibitor TLS](/mesosphere/dcos/1.14/security/ent/tls-ssl/exhibitor-tls) for more information. 
+
```

Comment:
When we initially discussed the project we talked about a graceful degradation vs. enforcing things. If I do not set `exhibitor_tls_enabled` then it defaults to `true`, and no graceful degradation is happening?

Reply:
> DC/OS global

"globally unique" is different from "DC/OS global, unique CA certificate". Which one is it?

Reply:
> the exhibitor service

Let's capitalize Exhibitor in this entire docs page.

Reply:
If `exhibitor_tls_enabled` is set to `true` and bootstrap service is not available then the bootstrap script does continue without enabling TLS authentication for exhibitor (degrades gracefully).

https://github.com/dcos/dcos/pull/5985/files#diff-eabe213470c95bef56238c23d4942061R179

Reply:
>  then the bootstrap script does continue without enabling TLS authentication for exhibitor (degrades gracefully).

Lovely. When we discussed this we said that this can be a security problem, and that there should be a way to prevent the graceful degradation from happening, i.e. to *enforce* things (to error out, if required). How can I enforce?

Reply:
The current implementation does not offer to enforce TLS authentication. Here I'd propose a new *boolean* configuration variable `exhibitor_tls_enforce` or `exhibitor_tls_required` that would allow enforcing Exhibitor TLS.

Reply:
> The current implementation does not offer to enforce TLS authentication

@jr0d is that true? 

From my email summarizing the initial discussion:

> If we do graceful degradation to not break the classical on-prem installation method then this might open up an attack vector. We might be able to improve the situation by introducing a new _optional_ DC/OS configuration parameter which enforces things.

Later on in the process I think I double-checked with you if we really build it that way, and I think I got your confirmation.

I think if this is supposed to introduce a security gain that operators can _rely_ upon ....

Reply:
@jgehrcke Yes, that is true, I will change the bootstrap code to exit with return code 1 when `exhibitor_tls_enabled` is `true`, the artifacts do not exist, and the bootstrap ca service is not available or there is an error during artifact generation.

Reply:
@jgehrcke I should also note that if we take this stance, we will need to make exhibitor_tls_enabled default to `false`. The reason being is that not all installation scenarios rely on the presence of a bootstrap node. At present, the existing e2e tests are passing due to graceful degradation, as an example.

Reply:
A little lost in the details, but I found our initial plan quite cool:
- by default, do the lock-down in a best-effort fashion (try to use it in as many cases as possible)
- if the operator says they absolutely require it and if it doesn't work, then error out during the installation.